File: 1874-minimize-product-sum-of-two-arrays/1874-minimize-product-sum-of-two-arrays.cpp
class Solution {
public:
    int minProductSum(vector<int>& nums1, vector<int>& nums2) 
    {
        sort(nums1.begin(), nums1.end());
        priority_queue <int, vector <int>> pq;
        for(int num : nums2)
            pq.push(num);
        int ans = 0;
        for(int idx = 0; idx < nums2.size(); idx++)
        {
            ans += nums1[idx] * pq.top();
            pq.pop();
        }
        return ans;
    }
};

File: 1874-minimize-product-sum-of-two-arrays/1874-minimize-product-sum-of-two-arrays.py
class Solution:
    def minProductSum(self, nums1: List[int], nums2: List[int]) -> int:
        
        nums1.sort()
        arr = [-num for num in nums2]
        heapq.heapify(arr)
        ans = 0
        
        for i in range(len(nums1)):
            ans += nums1[i] * -1 * heapq.heappop(arr)
        return ans
        